### ISSUE TITLE
More than one run with handler=hardlink deletes extended attributes

### ISSUE THREAD

**Issue by jim-collier** [sahib/rmlint#378]
## Steps

1. Dedupe directory:
    - `rmlint "/export/test_zfs" --no-crossdev --see-symlinks --hidden --xattr --types=none,duplicates --no-backup -o csv:/tmp/rmlint.csv -o json:/tmp/rmlint.json -o sh:/tmp/rmlint.sh --config=sh:handler=hardlink`
    - `/tmp/rmlint.sh -d -x`
1. Show the extended attributes that rmlint created for the files in that directory:

	```
	find /export/test_zfs -type f -print0 | while IFS= read -rd '' fileSpec; do
		getfattr -d "${fileSpec}" 2> /dev/null || true
	done
	```

1. Note that every file, including freshely hardlinked files, has `user.rmlint.blake2b.cksum` and `user.rmlint.blake2b.mtime` xattrs. (Also note that `user.rmlint.blake2b.mtime` for some duplicates appear to change, which is to be expected, because harlinked files share the same attrs & xattrs except for name.)
1. Clear out previous files: `rm /tmp/rmlint.*`
1. Run the same sequence of commands again starting with 1 above. __Notice that this time, there are no extended attributes__. Subsequent repeated calls doesn't recreate them, they "stay" deleted after the second run and beyond.

This is unexpected, and a problem when you need to run the command more than once on the same folder. Which, for custom or quick-&-dirty backups where you copy mostly the same data to, say, date-named subfolders on a dedicated "backup" machine, and rely on hardlinking to drastically reduce space by orders of magnitude. (And you need the xattrs to speed up subsequent processing and reduce disk grinding.)

Running rmlint with `--config=sh:handler=clone` multiple times, does not delete xattrs.

Running rmlint with these options recreates the xattrs (but running #1 again removes them again): `rmlint "/export/test_zfs" --no-crossdev --see-symlinks --hidden --xattr --types=none,duplicates,duplicatedirs --no-backup -o csv:/tmp/rmlint.csv -o json:/tmp/rmlint.json`

Fortunately, it doesn't delete other xattrs, such as user.com.dropbox.*.

## Build info

Compiled from master branch.

```
rmlint --version
version 2.9.0 compiled: Sep  4 2019 at [10:26:23] "Odd Olm" (rev 3a7d52db)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support
```

**Comment by sahib** [comment]
Hello @jim-collier,

sorry for the late response. There was a bug fix in bbc896f0dddb899d3321a4d5a0e2cbb96b18d043, which *could* be related to your problem. So far I was not able to reproduce the issue yet, but I only did a very quick and dirty test. Please try with newest ``develop`` again to see if it is better now or wait until I find some time to look more into it.

**Comment by SeeSpotRun** [comment]
HI @jim-collier 

I'm unable to reproduce this so will close issue if I don't hear back; assuming it was fixed somewhere along the way.

**Comment by jim-collier** [comment]
> sorry for the late response. There was a bug fix in [bbc896f](https://github.com/sahib/rmlint/commit/bbc896f0dddb899d3321a4d5a0e2cbb96b18d043), which _could_ be related to your problem. So far I was not able to reproduce the issue yet, but I only did a very quick and dirty test. Please try with newest `develop` again to see if it is better now or wait until I find some time to look more into it.

Thanks for responding eventually! Partially as a result of this and another bug (maybe?) or at least unexpected interaction with my script which was growing in complexity, I changed my "redundant data" strategy around that time from "deduplicate the redundant data", to "throw more hard drives at the problem".

I don't say that to try to make you guys "feel bad", it's a perfectly acceptable and frankly more fun workaround to, you know, actually solving my core problem of having too much redundant data.

But I do still intend to tackle the redundant data without actually removing the redundant data. That will follow me to the grave.) So I'll come back to this specific issue eventually! Perhaps in another three years.

**Comment by SeeSpotRun** [comment]
> perfectly acceptable and frankly more fun workaround to, you know, actually solving my core problem of having too much redundant data

Sounds a lot like the reason I got involved in this project in the first place lol.
